pkg/kn/commands/service_create_test.go

```diff
@@ -110,5 +110,58 @@ func TestServiceCreateEnv(t *testing.T) {
 		expectedEnvVars) {
 		t.Fatalf("wrong env vars %v", conf.RevisionTemplate.Spec.Container.Env)
 	}
+}
+
+func TestServiceCreateRequests(t *testing.T) {
+	action, created, _, err := fakeServiceCreate([]string{
+		"service", "create", "foo", "--image", "gcr.io/foo/bar:baz", "--requests-cpu", "250m", "--requests-memory", "64Mi"})
+
+	if err != nil {
+		t.Fatal(err)
+	} else if !action.Matches("create", "services") {
+		t.Fatalf("Bad action %v", action)
+	}
+
+	expectedRequestsVars := map[string]string{
+		"cpu":    "250m",
+		"memory": "64Mi"}
 
+	conf, err := servinglib.GetConfiguration(created)
+
+	if err != nil {
+		t.Fatal(err)
+	} else if conf.RevisionTemplate.Spec.Container.Image != "gcr.io/foo/bar:baz" {
+		t.Fatalf("wrong image set: %v", conf.RevisionTemplate.Spec.Container.Image)
+	} else if !reflect.DeepEqual(
+		conf.RevisionTemplate.Spec.Container.Resources.Requests,
+		expectedRequestsVars) {
+		t.Fatalf("wrong requests vars %v", conf.RevisionTemplate.Spec.Container.Resources.Requests)
+	}
+}
+
```

Comment:
Add a test for only specifying cpu, only specifying memory, and a test for a service update when you're changing memory but not cpu.

Reply:
Yeah, want to add more tests. The current ones not enough. Good point

Reply:
So to thoroughly test this we would need the following tests:

1. r CPU, r Mem (no limits)
2. r CPU only (no limits)
3. r Mem only (no limits)
4. l CPU, l Mem (no requests)
5. l CPU only (no requests)
6. l Mem only (no requests)
7. r CPU, l Mem only
8. l CPU, r Mem only

I believe that would cover all cases since the no limits and no requests is the test you had initially.

I think that's overkill but happy to churn those if needed, though I'd say this feature will then become the most tested of all the ones existing now...

Let me know your thoughts.

Reply:
OK since I see no strong opinion, I am adding something in the middle. Should cover a lot and we clearly need to have integration tests that cover more. I'll include this in my next commit.